libs/@local/hash-validation/src/data_type.rs

```diff
@@ -573,4 +695,63 @@ mod tests {
             .await
             .expect_err("validation succeeded");
     }
+
+    #[tokio::test]
+    async fn simple_mail() {
+        let mail_type = serde_json::to_string(&json!({
+            "$schema": "https://blockprotocol.org/types/modules/graph/0.3/schema/data-type",
+            "kind": "dataType",
+            "$id": "https://localhost:4000/@alice/types/data-type/simple-email/v/1",
+            "title": "E-Mail",
+            "type": "string",
+            "pattern": "^[^@]+@[^@]+$",
+        }))
+        .expect("failed to serialize simple mail type");
```

Comment:
I think it's confusing that we use an email to test `pattern` when an email schema should be expressed using `format: "email"` (and we don't test `format: "email"`) – can we pick some other regexp to test `pattern`?

Reply:
I changed it to `"Zip Code (US)"`, which is likely a good use for the pattern-property.